### ISSUE TITLE
NullPointerException on rotation when the picker is open

### ISSUE THREAD

**Issue by blunden** [ParkSangGwon/TedBottomPicker#43]
In the latest version the issue with crashing on rotation when the picker is open is back (previously reported in #19 and the same exception also happens in #42 ). The previous fix was seemingly reverted by this change: https://github.com/ParkSangGwon/TedBottomPicker/commit/c4f0907dfad838be6cc019e9cdeafda28022b5c0

04-15 12:28:30.079 3528-3528/CENSORED E/AndroidRuntime: FATAL EXCEPTION: main
    Process: CENSORED, PID: 3528
    java.lang.RuntimeException: Unable to start activity ComponentInfo{CENSORED/CENSORED.MainActivity}: java.lang.NullPointerException: Attempt to read from field 'boolean gun0912.tedbottompicker.TedBottomPicker$Builder.showTitle' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2666)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4520)
        at android.app.ActivityThread.-wrap19(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1484)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6128)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.NullPointerException: Attempt to read from field 'boolean gun0912.tedbottompicker.TedBottomPicker$Builder.showTitle' on a null object reference
        at gun0912.tedbottompicker.TedBottomPicker.setTitle(TedBottomPicker.java:483)
        at gun0912.tedbottompicker.TedBottomPicker.setupDialog(TedBottomPicker.java:168)
        at android.support.v4.app.DialogFragment.getLayoutInflater(DialogFragment.java:315)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
        at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2900)
        at android.support.v4.app.Fragment.performActivityCreated(Fragment.java:2207)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1326)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
        at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2900)
        at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:201)
        at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:603)
        at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:178)
        at CENSORED.MainActivity.onStart(MainActivity.java:162)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
        at android.app.Activity.performStart(Activity.java:6699)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2629)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4520) 
        at android.app.ActivityThread.-wrap19(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1484) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6128) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)

**Comment by ParkSangGwon** [comment]
@blunden 
Can you upload your sample project code?
I will test and check error.

**Comment by blunden** [comment]
I don't have a sample project so I would have to create one. I imagine that this happens in any app using this library though so if you have your own sample app it should be reproducible there too. If not, let me know and I'll have to create one.

You fixed this once before though as described above.

**Comment by rdadkins** [comment]
I faced this issue the other day. I believe it happens when the picker is opened and the underlying Fragment / Activity is destroyed and is trying to recreate the picker.

This may (more than likely) be caused by `onSaveInstanceState` not saving the `TedBottomPicker.Builder` and thus the NPE being thrown when trying to access `showTitle` on `builder` after recreating the Fragment. 

A solution would be to convert the Builder to a `Parcelable` and save it during `onSaveInstanceState` and restoring it when needed. The only issue would be that the listeners would not persist - this will require the library to change where listeners do not have to be provided in order to create a picker, instead they can be supplied later on.

A temporary solution is to save an instance of the picker dialog when it is shown and to `dismiss()` it during `onPause()`.

**Comment by iammert** [comment]
@rdadkins If you dismiss onPause,  camera intent result will not be handled by picker dialog and you won't get any callback.

**Comment by rdadkins** [comment]
@iammert You're right. I did not test this with the camera, only the initial gallery provided. The gallery intent will also not be handled from what I just discovered. I edited my original response.

**Comment by rdadkins** [comment]
Coming back to this issue, I found a pretty simple solution when using fragments (in Kotlin):

```
fun showTedBottomPicker() {
  val builder = getTedBottomPickerBuilder()
  ...
  val fragment = builder.create()
  fragment.show(fragmentManager, TedBottomPicker.TAG) // or any tag you want
}

fun getTedBottomPickerBuilder() = TedBottomPicker.Builder(context) // configure as needed

fun onCreateView(...) {
  super.onCreateView(...)
  fragmentManager.findFragmentByTag(TedBottomPicker.TAG)?.apply { // match tag above
    if (this is TedBottomPicker) {
      builder = getTedBottomPickerBuilder()
    }
  }
}
```

This will reassign all attributes including the title and click listeners that you define in `getTedBottomPickerBuilder()` and should prevent any future NPE's. 

There may be better places to utilize `findFragmentByTag` - I just found that `onCreateView` seems to work fine for me.

**Comment by sooorajjj** [comment]
@ParkSangGwon remove this https://github.com/ParkSangGwon/TedBottomPicker/blob/master/app/src/main/AndroidManifest.xml#L12
 and this issue is reproduceable everytime you rotate device to landscape .
Dirty fix for me ATM is to set it to Portrait in manifest.

**Comment by ParkSangGwon** [comment]
This issue solved at `v1.2.0`
